amespahdbpythonsuite/parsers.py

```diff
@@ -0,0 +1,543 @@
+#!/usr/bin/env python3
+"""
+parsers.py
+
+Parse a PAHdb XML file, with or without schema checking.
+"""
+
+import array
+import base64
+import os
+import urllib.request
+
+from lxml import etree as ElementTree
+import numpy as np
+
+
+class XMLparser:
+    """Parse a PAHdb XML file.
+
+    Optional behaviour includes validating the XML schema.
+
+    Attributes:
+        check_schema: Whether to validate the XML schema or not.
+        filename: XML file location. If not given, it is assigned to the
+            "AMESPAHDEFAULTDB" environment variable.
+
+    Note:
+        `validated` (bool) indicates if the schema is valid (True) or
+            not (False). Defaults to False until validation is performed.
+    """
+
+    def __init__(self, filename=None, check_schema=True):
+        """Inits XMLparser with schema checking on, no given filename."""
+        self.filename = filename
+        self.check_schema = check_schema
+        self.validated = False  # Assume schema has not yet been checked.
+
+        # Identify location of PAHdb XML file.
+        if self.filename is None:
+            try:
+                self.filename = os.environ["AMESPAHDEFAULTDB"]
+            except KeyError as e:
+                print("Must set AMESPAHDEFAULTDB environment variable.")
+                raise e
+            else:
+                print('Using default PAHdb file:\n{}'.format(self.filename))
+
+        # Parse XML file to dictionary.
+        self._database, self._info = self._parse()
+
+    def __repr__(self):
+        """Class representation."""
+        return (f'{self.__class__.__name__}('
+                f'check_schema={self.check_schema!r}, '
+                f'filename={self.filename!r}, '
+                f'validated={self.validated!r})')
+
+    def __str__(self):
+        """A description of the object."""
+        return (f'An XML parser from the Ames PAHdb Python Suite. '
+                f'XML file: \n{self.filename!r}.')
+
+    def _parse(self):
```

Comment:
This would be the 'to_dict'-method; lxml's parse would still be exposed as well. Perhaps 'to_dict' should be 'to_pahdb_dict' to make clear it is particular interpretation of the XML.